src/Orleans.TestingHost/TestClusterHostFactory.cs

```diff
@@ -0,0 +1,261 @@
+                    foreach (var source in configBuilder.Sources)
+                    {
+                        cb.Add(source);
+                    }
+                });
+
+            hostBuilder.Properties["Configuration"] = configuration;
+            ConfigureAppServices(configuration, hostBuilder);
+
+            hostBuilder.ConfigureServices((context, services) =>
+            {
+                services.AddSingleton<TestHooksSystemTarget>();
+                ConfigureListeningPorts(context, services);
+
+                TryConfigureTestClusterMembership(context, services);
+                TryConfigureFileLogging(configuration, services, siloName);
+
+                // TODO: make SiloHostBuilder work when not using the legacy configuration, similar to what we did with ClientBuilder.
+                // All The important information has been migrated to strongly typed options (everything should be migrated, but the minimum required set is already there).
+                var clusterConfiguration = GetOrCreateClusterConfiguration(services, context);
+                if (Debugger.IsAttached)
+                {
+                    // Test is running inside debugger - Make timeout ~= infinite
+                    clusterConfiguration.Globals.ResponseTimeout = TimeSpan.FromMilliseconds(1000000);
+                }
+            });
+
+            AddDefaultApplicationParts(hostBuilder.GetApplicationPartManager());
+
+            var host = hostBuilder.Build();
+            InitializeTestHooksSystemTarget(host);
+            return host;
+        }
+
+        public static IClusterClient CreateClusterClient(string hostName, IEnumerable<IConfigurationSource> configurationSources)
+        {
+            var configBuilder = new ConfigurationBuilder();
+            foreach (var source in configurationSources)
+            {
+                configBuilder.Add(source);
+            }
+            var configuration = configBuilder.Build();
+
+            var builder = new ClientBuilder();
+            builder.Properties["Configuration"] = configuration;
+            builder.ConfigureClusterClient(ob => ob.Bind(configuration));
+            ConfigureAppServices(configuration, builder);
+
+            builder.ConfigureServices(services =>
+            {
+                TryConfigureTestClusterMembership(configuration, services);
+                TryConfigureFileLogging(configuration, services, hostName);
+
+                if (Debugger.IsAttached)
+                {
+                    // TODO: configure this without requiring the legacy configuration, when that's available
+                    var clientConfiguration = GetOrCreateClientConfiguration(services, configuration);
+
+                    // Test is running inside debugger - Make timeout ~= infinite
+                    clientConfiguration.ResponseTimeout = TimeSpan.FromMilliseconds(1000000);
```

Comment:
Note: Once we move this to strongly typed options, we can get rid of the dependency to the legacy config objects in the TestClusterHostFactory

Reply:
Moved this to LegacyTestClusterConfiguration

Reply:
The reality is that this is useful in the non-legacy case too going forward. I would keep this by default, it's just that when you eventually move this setting out to a strongly typed option, then you can stop relying on the legacy configuration entirely